src/binary-reader-ir.cc

```diff
@@ -166,17 +166,18 @@ class BinaryReaderIR : public BinaryReaderNop {
   Result OnIfExpr(Type sig_type) override;
   Result OnLoadExpr(Opcode opcode,
                     Address alignment_log2,
-                    Address offset) override;
+                    Address offset,
+                    Index memidx) override;
```

Comment:
memidx appears right after the opcode in text format, but it appears after alignment and offset in binary format. I could update it such that for headers related to binary format, memidx comes last and for headers related to text format, memidx comes first, but I think it would be better if the ordering is the same for all headers.

Reply:
Agreed that they should all be the same, but I also think we should follow the text format and have memidx  come right after the opcode in all the APIs.   The fact that it comes last in the binary format is kind of an accident of history whereas the text format rightly puts it first where I think it makes more sense, no?

Reply:
The binary format uses bit 6 of alignment to indicate whether the current operation has a memidx immediate. Bit 6 is 1 if and only if there is a memidx immediate. Therefore, parser has to parse alignment first to figure out whether it is going to parse memidx, and that's why memidx is placed at the last. But I agree that it makes more logical sense to put memidx right after opcode. I will update the APIs.